cypress/integration/unit_tests/add_disaster_test.js

```diff
@@ -0,0 +1,184 @@
+import {gdEeStatePrefix, legacyStateDir, legacyStatePrefix} from '../../../docs/ee_paths.js';
+import {getFirestoreRoot} from '../../../docs/firestore_document.js';
+import {addDisaster, createAssetPickers, createOptionFrom, disasters, emptyCallback, getAssetsFromEe, stateAssets, writeNewDisaster} from '../../../docs/import/add_disaster.js';
+import {addFirebaseHooks, loadScriptsBeforeForUnitTests} from '../../support/script_loader.js';
+
+const KNOWN_STATE = 'WF';
+const UNKNOWN_STATE = 'DN';
+const KNOWN_STATE_ASSET = gdEeStatePrefix + KNOWN_STATE + '/snap';
+
+describe('Unit tests for add_disaster page', () => {
+  loadScriptsBeforeForUnitTests('ee', 'firebase', 'jquery');
+  addFirebaseHooks();
+  before(() => {
+    cy.wrap(firebase.auth().signInWithCustomToken(firestoreCustomToken));
+
+    const disasterPicker = createAndAppend('select', 'disaster');
+    disasterPicker.append(createOptionFrom('2001-summer'));
+    disasterPicker.append(createOptionFrom('2003-spring'));
+    disasterPicker.val('2003-spring');
+
+    createAndAppend('div', 'status').hide();
+  });
+
+  beforeEach(() => {
+    const listAssetsStub = cy.stub(ee.data, 'listAssets');
+    listAssetsStub.withArgs(legacyStateDir, {}, emptyCallback)
+        .returns(Promise.resolve({
+          'assets': [{
+            id: gdEeStatePrefix + KNOWN_STATE,
+          }],
+        }));
+    listAssetsStub.withArgs(legacyStatePrefix + KNOWN_STATE, {}, emptyCallback)
+        .returns(Promise.resolve({
+          'assets': [
+            {
+              id: gdEeStatePrefix + KNOWN_STATE + '/snap',
+              type: 'TABLE',
+            },
+            {
+              id: gdEeStatePrefix + KNOWN_STATE + '/folder',
+              type: 'FOLDER',
+            },
+          ],
+        }));
+    cy.stub(ee.data, 'createFolder');
+
+    stateAssets.clear();
+    // In prod this would happen in enableWhenReady which would read from
+    // firestore.
+    disasters.clear();
+    disasters.set('2001-summer', []);
+    disasters.set('2003-spring', []);
+  });
+
+  afterEach(() => {
+    stateAssets.clear();
+    disasters.clear();
+  });
+
+  it('gets state asset info from ee', () => {
+    getAssetsFromEe([KNOWN_STATE, UNKNOWN_STATE]).then((assets) => {
```

Comment:
can you cy.wrap() this first promise (you can leave the then() unchanged). If this promise never resolved, the test would actually pass without that. Here and below.